### PR TITLE
Updates/frontier v6

### PR DESCRIPTION
**Pull request by RedRafe**
Round of updates for the Frontier scenario

- [x] added artillery turrets to the Enemy turrets feature (will auto target players and structures when registering such turret)
- [x] new terrain generation
- [x] fixed bugs with silo positioning on invalid tiles and filling fluid enemy turrets

### CODE REVIEW

**Comment by RedRafe** [comment]
There are bits commented out, especially in the debug modules for terrain gen. I'm not in a hurry to merge this but I wanted to get it out as next updates will include some refactors/additions in other modules outside Frontier map.
So maybe this could also be considered a draft before I come back to it, just because I prefer to keep the code on git updated with the one currently running on serve, or at least have a copy of it for reference.

File: features/enemy_turret.lua [hunk 1]
```diff
@@ -75,6 +134,65 @@ function Public.reset()
...
+
+  if #targets == 0 then
+    return
+  end
+
+  for i = 1, #targets do
+    local target = targets[i]
+    if distance(source_position, target.position) > Artillery.min_range then
+      Task.set_timeout_in_ticks(Artillery.fire_interval, artillery_projectile_token, {
+        surface_index = surface.index,
```
Comment (grilledham) [inline features/enemy_turret.lua]: Minor, but you could pass the surface rather than the surface.index.

**Review by grilledham — APPROVED** [review]
I'd be happy to merge in it's current form. But I'll wait for you to say you are ready.

File: features/enemy_turret.lua (hunk 1 above)
Comment (RedRafe) [reply features/enemy_turret.lua]: Yup, thanks

**Comment by RedRafe** [comment]
Tidied up a little. This one can go in now

**Review by grilledham — APPROVED** [review]

**Pull request merged**